examples/hal/quad/main.rs

```diff
@@ -621,3 +519,206 @@ fn main() {
 fn main() {
     println!("You need to enable the native API feature (vulkan/metal) in order to test the LL");
 }
+
+#[cfg(any(feature = "vulkan", feature = "dx12", feature = "metal", feature = "gl"))]
+fn swapchain_stuff(
+    surface: &mut <back::Backend as hal::Backend>::Surface,
+    device: &mut back::Device,
+    set_layout: &<back::Backend as hal::Backend>::DescriptorSetLayout,
+) -> (
+    <back::Backend as hal::Backend>::Swapchain,
+    <back::Backend as hal::Backend>::RenderPass,
+    std::vec::Vec<<back::Backend as hal::Backend>::Framebuffer>,
+    std::vec::Vec<(
+        <back::Backend as hal::Backend>::Image,
+        <back::Backend as hal::Backend>::ImageView,
+    )>,
+    <back::Backend as hal::Backend>::GraphicsPipeline,
+    <back::Backend as hal::Backend>::PipelineLayout,
+    Extent2D,
+) {
+    let surface_format = surface
+        .capabilities_and_formats(device.get_physical_device())
```

Comment:
Let's follow the same logic as a Vulkan application:
  - pass in the physical device here that is held by the user
  - don't consume it on `open`